apps/next15-demo/src/app/StoryblokComponent.tsx

```diff
@@ -0,0 +1,134 @@
+        {JSON.stringify(blok, null, 2)}
+      </pre>
+    </div>
+  );
+};
+
+type Component =
+  | NonNullable<StoryblokComponents.DefaultPage["body"]>[number]
+  | StoryblokComponents.DefaultPage
+  | StoryblokComponents.HeadlineSegment
+  | StoryblokComponents.Button
+  | StoryblokComponents.NavItem
+  | StoryblokComponents.ArticlePage
+  | StoryblokComponents.ArticleOverviewPage
+  | StoryblokComponents.Category
+  | StoryblokComponents.NewsletterFormSection
+  | StoryblokComponents.ContactFormSection
+  | StoryblokComponents.PriceCard
+  | StoryblokComponents.GridCard
+  | StoryblokComponents.ImageCard
+  | StoryblokComponents.RichtextYoutube
+  | StoryblokComponents.TabbedContentEntry;
+
+export const StoryblokComponent = ({
+  blok,
+  story,
+  ...props
+}: {
+  blok: Component;
+  story?: ISbStoryData;
+  [prop: string]: unknown | undefined;
+}) => {
+  const { Blok, Bloks } = useBloks<Component>();
+
+  return (
+    <Bloks {...props} blok={blok} story={story} fallback={FallbackComponent}>
+      <Blok
+        component="featured-articles-section"
+        element={FeaturedArticlesSection}
+      />
+      <Blok component="grid-section" element={GridSection} />
+      <Blok component="hero-section" element={HeroSection} />
+      <Blok component="image-text-section" element={ImageTextSection} />
+      <Blok component="tabbed-content-section" element={TabbedContentSection} />
+      <Blok component="text-section" element={TextSection} />
+      <Blok component="banner" element={Banner} />
+      <Blok component="banner-reference" element={BannerReference} />
+      <Blok
+        component="latest-articles-section"
+        element={LatestArticlesSection}
+      />
+      <Blok component="faq-section" element={FaqSection} />
+      <Blok component="logo-section" element={LogoSection} />
+      <Blok
+        component="newsletter-form-section"
+        element={NewsletterFormSection}
+      />
+      <Blok component="personalized-section" element={PersonalizedSection} />
+      <Blok component="products-section" element={ProductsSection} />
+      <Blok component="testimonials-section" element={TestimonialsSection} />
+      <Blok component="article-overview-page" element={ArticleOverviewPage} />
+      <Blok component="article-page" element={ArticlePage} />
+      <Blok component="category" element={Category} />
+      <Blok component="contact-form-section" element={ContactFormSection} />
+      <Blok component="price-card" element={PriceCard} />
+      <Blok component="grid-card" element={GridCard} />
+      <Blok component="image-card" element={ImageCard} />
+      <Blok component="richtext-youtube" element={RichtextYoutube} />
+      <Blok component="tabbed-content-entry" element={TabbedContentEntry} />
+      <Blok component="headline-segment" element={HeadlineSegment} />
+      <Blok component="default-page" element={DefaultPage} />
+      <Blok component="nav-item" element={NavItem} />
+    </Bloks>
```

Comment:
This the crux of the API. It's basically a gloried switch statement. This is user generated code and users can create as many or as few of these as they choose to target different rendering contexts.